### DISCUSSION TITLE
pinocchio::appendModel does not really append

### DISCUSSION THREAD

**Discussion by florent-lamiraux** [stack-of-tasks/pinocchio#2203] [General]
When calling function `pinocchio::appendModel`, with `model1`, `model2` and `frame` (of model1 to which model2 should be attached), model2 is added to model1 using the following rule:
  1. add joints of `model1` up to the parent joint of `frame`,
  2. add model2 as a child of the latter joint,
  3. add the remaining joints of `model1`.
  As a consequence, when `frame` is `universe`, `model2` is placed before `model1` in the kinematic chain. This is rather unintuitive. Is there a reason for this ?

I would expect
1. add joints of `model1`,
2. add joints of `model2`.
which is much simpler and still satisfies that any joint comes after its parent.

**Comment by jmirabel** [comment]
Note that joint ordering must respect compactness. All the children of a joint follows that joint in the list of joints. Between a joint J and its furthest child (in the list of joints), there cannot be a joint that is not a child of joint J.

So in the general case, you cannot just append one model after the other. However, it makes more sense to append all the joints of model1 until the last child of the parent joint of `frame`, then model2 and then the rest of model1.
